app/models/product_importer.rb

```diff
@@ -177,7 +181,12 @@ def count_existing_products
           AND spree_variants.deleted_at IS NULL', supplier_id).
           count
 
-        @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        if @supplier_products[:by_supplier][supplier_id]
+          @supplier_products[:by_supplier][supplier_id][:existing_products] = products_count
+        else
+          @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        end
```

Comment:
I see, maybe storing the whole state of this transaction in a Hash is not the most flexible way and can lead easily to future bugs overriding keys. Imagine that I don't notice that detail and I make a PR removing that `if`...

I generally use ruby objects to represent this kind of state. Another thing that you gain by using ruby objects is that you can encapsulate logic in those objects making this `ProductImporter` much easier to follow, maintain and test.

Of course doing that now would imply dedicating more time on this so don't bother me... but maybe we can just talk about it and see how this feature could be refactored using ruby objects instead of Hash structures. Ping me any time if / when you want to do this. 💪